src/jrd/tpc.cpp

```diff
@@ -772,16 +771,38 @@ SnapshotHandle TipCache::beginSnapshot(thread_db* tdbb, AttNumber attachmentId,
 	// Remap snapshot list if it has been grown by someone else
 	remapSnapshots(false);
 
+	if (commitNumber != 0)
+	{
+		SnapshotList* snapshots = m_snapshots->getHeader();
+		ULONG slotsUsed = snapshots->slots_used.load(std::memory_order_relaxed);
+		bool found = false;
+
+		for (SnapshotHandle slotNumber = 0; slotNumber < slotsUsed; ++slotNumber)
+		{
+			if (snapshots->slots[slotNumber].attachment_id.load(std::memory_order_relaxed) != 0 &&
+				snapshots->slots[slotNumber].snapshot.load(std::memory_order_relaxed) == commitNumber)
+			{
+				found = true;
+				break;
+			}
+		}
+
+		if (!found)
+			ERR_post(Arg::Gds(isc_tra_snapshot_does_not_exist));
+	}
```

Comment:
Probably, it makes sense to add value of missing snapshot number into error message

Reply:
As string? isc_arg_number is 32-bit...

Reply:
The same way as isc_concurrent_transaction reports transaction number.
Hmm... is it broken ?

Reply:
I believe it is really broken now with 64-bit transaction numbers.

Reply:
I do not understand why `Num::Num` constructor accepts an intptr_t instead of proper 32 bits.

Reply:
Because all items put into status-vector array ?

Reply:
But the constructor parameter does not need to be. But anyway, it does not seems to catch the 64 bits as a warning when changing it.

Reply:
Time to introduce `isc_arg_number2` ? :)